src/rviz/visualizer_app.cpp

```diff
@@ -221,6 +221,13 @@ bool VisualizerApp::init( int argc, char** argv )
     ros::NodeHandle private_nh("~");
     reload_shaders_service_ = private_nh.advertiseService("reload_shaders", reloadShaders);
 
+    //process any callbacks that don't use the designated callback queues of VisualizationManager
```

Comment:
Why do you want to handle the global queue differently?

Reply:
`VisualizerApp` is used by the rviz binary but not by library users. Handling the global queue here will make sure that plugins that rely on it won't see any difference. For library users it allows not handling the global queue in the main thread.